File: ExampleApp/src/Sink.cs
using Prism;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Example
{
    class Sink : Behaviour
    {

        public float SinkSpeed;

        void OnCreate()
        {

        }

        void OnUpdate()
        {
            float ts = Time.DeltaTime;
            Matrix4 transform = Entity.GetTransform();
            Vector3 translation = transform.Translation;

            translation.Y -= SinkSpeed * ts;

            transform.Translation = translation;
            Entity.SetTransform(transform);
        }

    }
}

File: PrismEditor/Assets/scripts/Python/Sink.py
from Prism import Behaviour, Time
from Prism.Math import Vector3


class Sink(Behaviour):
    SinkSpeed: float = 0.0

    def OnCreate(self):
        pass

    def OnUpdate(self):
        ts = Time.DeltaTime

        pos = self.Entity.Transform.Position
        pos.y -= self.SinkSpeed * ts
        self.Entity.Transform.Position = pos
